### PR TITLE
PD-751: Make existence checks include null

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- Make existence checks include null
= This means that if a user passes in null it'll be treated as not existing.

🎟 _Jira ticket:_ [PD-751](https://jira.mongodb.org/browse/PD-751)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/box/src/Box.tsx [hunk 1]
```diff
@@ -39,13 +39,13 @@ function InlineBox<TProps>(
...
 function InlineBox(props: BoxProps, ref: React.Ref<any>) {
-  if (props.as !== undefined) {
+  if (props.as != null) {
     const { as: Component, ...rest } = props;
     // @ts-expect-error
     return <Component {...rest} ref={ref} />;
   }
 
-  if (props.href !== undefined) {
+  if (props.href != null) {
```
Comment (lazytype) [inline packages/box/src/Box.tsx:48]: can we go even so far as to make this`typeof props.href !== 'string'`?

Comment (brookescarlett) [reply packages/box/src/Box.tsx:48]: wfm

Comment (brookescarlett) [reply packages/box/src/Box.tsx]: actually, getting this from ts

```Types of property 'onCopy' are incompatible.
        Type '((event: ClipboardEvent<HTMLAnchorElement>) => void) | undefined' is not assignable to type '((event: ClipboardEvent<HTMLDivElement>) => void) | undefined'.
          Type '(event: ClipboardEvent<HTMLAnchorElement>) => void' is not assignable to type '(event: ClipboardEvent<HTMLDivElement>) => void'.
            Types of parameters 'event' and 'event' are incompatible.
              Type 'ClipboardEvent<HTMLDivElement>' is not assignable to type 'ClipboardEvent<HTMLAnchorElement>'.
                Type 'HTMLDivElement' is missing the following properties from type 'HTMLAnchorElement': charset, coords, download, hreflang, and 21 ```

Comment (brookescarlett) [reply packages/box/src/Box.tsx]: I can enforce `string` in other components (Menu, SubMenu and Button) but not in Box 🤔

Comment (lazytype) [reply packages/box/src/Box.tsx:48]: my guess is that
```js
if (typeof props.href != 'string') {
  return ...
}
// props is not being narrowed from `BoxAnchorDefault | BoxDivDefault` to `BoxDivDefault` anymore
```
but no biggie

File: packages/button/src/Button.spec.tsx [hunk 2]
```diff
@@ -61,6 +61,11 @@ describe('packages/button', () => {
     expect(button.tagName.toLowerCase()).toBe('a');
   });
 
+  test(`renders component inside of button tag when "href" prop is set to null`, () => {
+    const { button } = renderButton({ href: null });
+    expect(button.tagName.toLowerCase()).toBe('button');
```
Comment (hswolff) [inline packages/button/src/Button.spec.tsx:66]: nice!!!

**Pull request merged**